src/osp/Resource/resourcetypes.h

```diff
@@ -93,19 +94,40 @@ struct res_container
 template <typename T>
 using res_container_t = typename res_container<T>::type;
 
+// Resource Owner
+
 class Resources;
 
-using ResIdStorage_t = lgrn::IdStorage<ResId, Resources>;
+using ResIdOwner_t = lgrn::IdStorage<ResId, Resources>;
+
+
+// Resource Type Ids
+
+struct GenResTypeId
+{
+    friend inline ResTypeId resource_type_next() noexcept;
+    friend inline int resource_type_count() noexcept;
+private:
+    static inline int count{0};
```

Comment:
Is int really the right choice?

At the least, shouldn't it be ResTypeId ?

Reply:
Mentioned above, this number isn't going to go that high. It would still need to be casted to an integer type to increment. I could use the underlying type, but that excess complication.